### PR TITLE
api: test migration 23

### PR DESCRIPTION
**Pull request by eduardoveiga**

### CODE REVIEW

File: api/pkg/dbtest/dbserver.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func (dbs *DBServer) start() {
 	}
 
 	args := []string{
-		"run", "--rm", fmt.Sprintf("--net=%s", dbs.network), "mongo:4.2.8",
+		"run", "--rm", fmt.Sprintf("--net=%s", dbs.network), "mongo:4.4.4",
```
Comment (otavio) [inline api/pkg/dbtest/dbserver.go:106]: Please split this for a new PR as this is not related.

File: api/store/mongo/migrations/migration_23_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		migration_14,
+		migration_15,
+		migration_16,
+		migration_17,
+		migration_18,
+		migration_19,
+		migration_20,
+		migration_21,
+		migration_22,
+	}
```
Comment (otavio) [inline api/store/mongo/migrations/migration_23_test.go:42]: I believe every migration test will require this very same logic. We might try to come up with would be a method such as: `migration.GetMigrationsUntil(23)` and `migration.GetNextMigration()` to avoid this duplication in every migration test.

How does it sound?

Comment (gustavosbarreto) [reply api/store/mongo/migrations/migration_23_test.go:42]: https://play.golang.org/p/EChLMNf_RrP

File: api/store/mongo/migrations/migration_23_test.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	migrate "github.com/xakep666/mongo-migrate"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+func TestMigration23(t *testing.T) {
+	db := dbtest.DBServer{}
+	defer db.Stop()
+
+	migrations := []migrate.Migration{
```
Comment (gustavosbarreto) [inline api/store/mongo/migrations/migration_23_test.go:19]: migrations := GenerateMigrations()[:22]

File: api/store/mongo/migrations/migration_23_test.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+		Identity: &models.DeviceIdentity{MAC: "mac"},
+		TenantID: "tenant2",
+		LastSeen: time.Now(),
+	}
+	if _, err := db.Client().Database("test").Collection("devices").InsertOne(context.TODO(), device); err != nil {
+		t.Errorf("failed to insert document")
+		return
+	}
+
+	migrations = []migrate.Migration{
```
Comment (gustavosbarreto) [inline api/store/mongo/migrations/migration_23_test.go:105]: migrations = GenerateMigrations()[22] // migration version 23 (NOTE: zero-based numbering)

File: api/store/mongo/migrations/migration_23_test.go (hunk 2 above)
Comment (otavio) [reply api/store/mongo/migrations/migration_23_test.go:42]: I thought about something like: https://play.golang.org/p/5G6lPgdAwnk

@gustavosbarreto  what do you think?

**Review by gustavosbarreto — APPROVED** [review]

**Review by gustavosbarreto — CHANGES_REQUESTED** [review]
@eduardoveiga please rebase

**Review by gustavosbarreto — APPROVED** [review]

**Comment by gustavosbarreto** [comment]
LGTM
